torch/utils/tensorboard/summary.py

```diff
@@ -518,7 +518,7 @@ def audio(tag, tensor, sample_rate=44100):
     wave_write.setnchannels(1)
     wave_write.setsampwidth(2)
     wave_write.setframerate(sample_rate)
-    wave_write.writeframes(tensor.data)
+    wave_write.writeframes(tensor.detach())
```

Comment:
Indeed. We could fix that by calling `make_np(tensor)`. This approach is already used in [`video(...)`](https://github.com/pytorch/pytorch/blob/f0ada4bd54d24f4bd50fd7d2f101fd4c472446d9/torch/utils/tensorboard/summary.py#L456). If that's ok, I can update.

Reply:
Actually `.data` is a valid thing on numpy array: https://numpy.org/doc/stable/reference/generated/numpy.ndarray.data.html
So this one I think we just want to rename `tensor` to `array` and keep the `.data`!

Reply:
I didn't discuss that but, although `.data` is valid, it is not required in this case. It returns a `memoryview`, and `id(memoryview(x)) == id(x.data)`. As `wave` gets a `memoryview` in both cases (see [here](https://github.com/python/cpython/blob/3e3ff09058a71b92c32d1d7dc32470c0cf49300c/Lib/wave.py#L424-L425)), and the test without `.data` passes, I guess we could simply remove it. Anyway, I think we can proceed as you're suggesting, because at this point, given `make_np` call at the beginning, it'll be a numpy array. I'll submit the change.